src/main/java/org/traccar/protocol/RuptelaProtocolDecoder.java

```diff
@@ -236,7 +237,17 @@ protected Object decode(
                 buf.readUnsignedByte(); // timestamp extension
 
                 if (type == MSG_EXTENDED_RECORDS) {
-                    buf.readUnsignedByte(); // record extension
+                    int recordExtension = buf.readUnsignedByte(); // record extension
+                    int noRecordsToMerge = BitUtil.between(recordExtension, 4, 8);
+                    int currentRecord = BitUtil.to(recordExtension, 4);
+
+                    if (currentRecord > 0 && noRecordsToMerge >= currentRecord) {
```

Comment:
Reads better this way

```suggestion
                    if (currentRecord > 0 && currentRecord <= noRecordsToMerge) {
```